internal/builders/golang/build.go

```diff
@@ -182,6 +187,9 @@ func checkMain(build config.Build) error {
 	if main == "" {
 		main = "."
 	}
+	if build.Dir != "" {
+		main = filepath.Join(build.Dir, main)
```

Comment:
I'm not sure why this is necessary... can you explain?

Reply:
This is necessary to enable GoReleaser to search for the main function in the right directory.

For example, if Go code is in project's `go/` subdirectory and main is in `foo/main.go`, then without this, the next line:
https://github.com/goreleaser/goreleaser/blob/468401bc54d3a31aebcc1526ba803760a80b51c3/internal/builders/golang/build.go#L185

would wrongly stat the `foo/main.go` file and return:
```
error=stat ./foo/main.go: no such file or directory
```

With this change, the code correctly stats the `go/foo/main.go` file.